blog/2022-02-15-how-datadog-ensures-database-reliability-with-temporal.md

```diff
@@ -0,0 +1,111 @@
+
+## **Migration**
+
+As a first step in the migration, runbooks were converted to workflows, removing the need for a human to tie together or even execute individual steps. This greatly reduced the chance of drift between different systems and teams. As an example*, if a runbook previously expected a human to run an operation on a Kubernetes cluster, the Temporal Workflow would instead execute an Activity which runs a K8s command on the cluster*
+
+> For most of the runbooks we already had standalone APIs or tools that we could easily leverage within Temporal Workflows and Activities. For example, instead of having the runbook that tells you "Please run this command on this cluster?" What we'll do is write an Activity that will create a Kubernetes job that runs a command. This was quite easy to do.
+
+As things moved forward the DRE team began to realize the hardest part of the migration was building trust in the new system, simply because it was new. While the original runbook approach was messy, the manual nature meant that if something went wrong an operator was already there to detect and react to it. When the team first started running Workflows they felt anxious because of basic questions:
+
+- How do I pause it?
+- How do I stop it?
+- If things are going wrong how do I detect it?
+
+> Before Temporal, we knew how to do things, albeit manually. If something bad happened, we could react to it, detect it and remediate. But when team members start a workflow (especially for the first few times), there are always these questions of... How do I pause it? How do I stop it? If things are going wrong, how will I detect it?
+
+It took some time for the team to trust the new Temporal solution. For a while they even ran the new system in parallel with the old one. Eventually this created enough trust that the team felt safe completely moving to the Temporal based solution.
+
+## **Outcomes**
+
+The initial goal of the migration was to move from almost completely manual runbooks to automated and highly reliable Workflows. Once the team developed trust in the Temporal solution, they almost immediately started seeing benefits of the switch. Temporal provided DRE the “glue” to tie together all of the critical work they are constantly doing behind the scenes
+
+> We had close to nothing. We had runbooks, bash for loops and what we call toolboxes, which is like a pod running in Kubernetes next to our cluster on which we can run commands. Temporal was filling a giant gap in our operations.
+
+Automating things not only improved reliability and continuity, it also improved productivity since team members no longer needed to sit and wait for long running operations to run. They just let the Workflow handle it.
```

Comment:
```suggestion
Automating things not only improved reliability and continuity, it also improved productivity because team members no longer needed to sit and wait for long-running operations to run. They just let the Workflow handle it.
```